addon/components/tf-button/component.js

```diff
@@ -0,0 +1,35 @@
+/**
+  @module ticketfly-buttons
+ */
+import Ember from 'ember';
+import layout from './template';
+import { collect } from 'ember-computed';
+const { Component, computed, get } = Ember;
+
+/**
+  @public
+  @class TfButton
+  @extends Ember.Component
+ */
+export default Component.extend({
+  layout,
+
+  tagName: 'button',
+  classNames: ['c-button'],
+  attributeBindings: ['disabled'],
+  classNameBindings: ['buttonCategory'],
+
+  utilityGroups: collect('weightGroup', 'colorGroup'),
+
+  buttonCategory: computed('utilityGroups', {
+    get() {
+      let buttonCategory = '';
+      get(this, 'utilityGroups').map((group) => {
+        if (group) {
+          buttonCategory += `u-button-${group} `;
```

Comment:
@BrianSipple / @melissaroman — any thoughts? 

I've literally _never_ used BEM, so I have no context to have an opinion. :-P 

The only question I have is wether we need to think of an `tf-icon-button` and a `tf-button` as separate components. So, if you used the inline form of the component, then you'd get:

```html
<button class="tf-button tf-button--secondary tf-button--neutral">
  <span>Click Me</span>
</button>
```

And, if it had an icon, the HTML would be your second example:

```html
<button class="tf-button tf-button--secondary tf-button--neutral">
  <span class="tf-button__icon-plus"></span>
  <span>Add a thing</span>
</button>
```